src/components/modal/index.tsx

```diff
@@ -91,3 +101,10 @@ const styles = StyleSheet.create({
 });
 
 Modal.TopBar = TopBar;
+
+export default asBaseComponent<
+  ModalPropTypes,
+  {
+    TopBar: typeof TopBar;
+  }
```

Comment:
Consider changing to `typeof Modal` it will automatically pass the type of Modal class with all of its static member. 
This is what we do in our Button component, let's see if it works here as well.

Reply:
Done